Signal/src/util/ThreadUtil.m

```diff
@@ -84,6 +88,112 @@ + (void)sendMessageWithAttachment:(SignalAttachment *)attachment
+                              TSIncomingMessage *incomingMessage = (TSIncomingMessage *)object;
+                              if (!firstIncomingMessage ||
+                                  [[firstIncomingMessage receiptDateForSorting]
+                                      compare:[incomingMessage receiptDateForSorting]]
+                                      == NSOrderedDescending) {
+                                  firstIncomingMessage = incomingMessage;
+                              }
+                          } else if ([object isKindOfClass:[TSOutgoingMessage class]]) {
+                              TSOutgoingMessage *outgoingMessage = (TSOutgoingMessage *)object;
+                              if (!firstOutgoingMessage ||
+                                  [[firstOutgoingMessage receiptDateForSorting]
+                                      compare:[outgoingMessage receiptDateForSorting]]
+                                      == NSOrderedDescending) {
+                                  firstOutgoingMessage = outgoingMessage;
+                              }
+                              outgoingMessageCount++;
+                              if (outgoingMessageCount > kMaxOutgoingMessageCount) {
+                                  // If the user has sent more than N interactions, abort.
+                                  *stop = YES;
+                              }
+                          }
+                      }];
+
+        if (!firstIncomingMessage && !firstOutgoingMessage) {
+            // If the thread has no interactions, abort.
+            return;
+        }
+
+        if (outgoingMessageCount > kMaxOutgoingMessageCount) {
+            // If the user has sent more than N messages, abort.
+            return;
+        }
+
+        if (hasUnknownContactBlockOffer) {
+            // If there already is a block offer, abort.
+            return;
+        }
+
+        BOOL hasOutgoingBeforeIncomingInteraction = (firstOutgoingMessage
+            && (!firstIncomingMessage ||
+                   [[firstOutgoingMessage receiptDateForSorting] compare:[firstIncomingMessage receiptDateForSorting]]
+                       == NSOrderedAscending));
+        if (hasOutgoingBeforeIncomingInteraction) {
+            // If there is an outgoing message before an incoming message
+            // the local user initiated this conversation, abort.
+            return;
+        }
+
+        DDLogInfo(@"Creating block offer for unknown contact");
+
+        // We want the block offer to be the first interaction in their
+        // conversation's timeline, so we back-date it to slightly before
+        // the first incoming message (which we know is the first message).
+        TSIncomingMessage *firstMessage = firstIncomingMessage;
+        uint64_t blockOfferTimestamp = firstMessage.timestamp - 1;
+
+        // Create an error message we may or may not user.
```

Comment:
> Create an error message we may or may not use

clarity: What does this mean? Maybe it would make sense if I know who "we" is - the app or the user?